platform/commonUI/browse/src/navigation/NavigateAction.js

```diff
@@ -47,36 +44,12 @@ define(
          *          navigation has been updated
          */
         NavigateAction.prototype.perform = function () {
-            var self = this,
-                navigateTo = this.domainObject,
-                currentObject = self.navigationService.getNavigation();
-
-            function allow() {
-                var navigationAllowed = true;
-                self.policyService.allow("navigation", currentObject, navigateTo, function (message) {
-                    navigationAllowed = self.$window.confirm(message + "\r\n\r\n" +
-                        " Are you sure you want to continue?");
-                });
-                return navigationAllowed;
-            }
-
-            function cancelIfEditing() {
-                var editing = currentObject.hasCapability('editor') &&
-                    currentObject.getCapability('editor').isEditContextRoot();
-
-                return self.$q.when(editing && currentObject.getCapability("editor").finish());
-            }
-
-            function navigate() {
-                return self.navigationService.setNavigation(navigateTo);
-            }
-
-            if (allow()) {
-                return cancelIfEditing().then(navigate);
-            } else {
-                return this.$q.when(false);
+            if (this.navigationService.shouldNavigate()) {
```

Comment:
The one thing that I'm feeling iffy about here is the addition of the `shouldNavigate` method to NavigationService. I think it would be non-obvious during maintenance which `setNavigation` calls should  depend upon this check and which should not.

Is there any reason not to include this check/prompt as part of the `setNavigation` call? `setNavigation` could also return the promise that's needed here (rejected if the navigation does not occur for some reason), which would in turn make it simpler to change the prompt from a `window.confirm` to a custom asynchronous dialog at a later time.